mass_mailing_partner/models/res_partner.py

```diff
@@ -31,12 +30,12 @@ class ResPartner(models.Model):
     @api.constrains('email')
     def _check_email_mass_mailing_contacts(self):
         for partner in self:
-            if partner.sudo().mass_mailing_contact_ids and not partner.email:
-                raise ValidationError(
-                    _("This partner '%s' is subscribed to one or more "
-                      "mailing lists. Email must be assigned.") % partner.name)
+            if not partner.email and self.sudo().mass_mailing_contact_ids:
+                raise ValidationError(_(
+                    "This partner '%s' is linked to one or more mass "
+                    "mailing contact. Email must be assigned."
```

Comment:
I'm thinking that it should discriminate when mass mailings are in queue. Otherwise, there would be no problem in removing the email, right?

Reply:
But the field email in the mass_mailing contact model is still required. Setting the linked partner email to blank would raise an exception.